## In Class Practice with Pandas

**Data Dictionary**

* Name - Name of the person
* Net worth in billions - Net worth of the person in billions
* Country - Country where the person is from
* Source - Source of the income
* Rank - Rank of the person in the Billionaires list
* Age - Age of the person
* Industry - Industry to which the person is related to

### Importing Pandas Library

In [1]:
import pandas as pd

### Loading the dataset in a pandas dataframe

In [2]:
df = pd.read_csv('Forbes-list-2020.csv')

### Print top 5 rows

In [3]:
df.head(5)

Name  NetWorth in billions        Country  \
0                Jeff Bezos                 177.0  United States   
1                 Elon Musk                 151.0  United States   
2  Bernard Arnault & family                 150.0         France   
3                Bill Gates                 124.0  United States   
4           Mark Zuckerberg                  97.0  United States   

          Source  Rank  Age          Industry  
0         Amazon     1   57        Technology  
1  Tesla, SpaceX     2   49        Automotive  
2           LVMH     3   72  Fashion & Retail  
3      Microsoft     4   65        Technology  
4       Facebook     5   36        Technology

#### <span style="color:crimson">Q1. Whose name is assigned to the 0th index in the dataframe? Jeff Bezos</span>

### Print the dimensions of data

In [4]:
df.shape

(2755, 7)

#### <span style="color:crimson">Q2. How many records do you have in this dataset? 2755</span>

### Datatypes of all the fields in the data

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2755 entries, 0 to 2754
Data columns (total 7 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Name                  2755 non-null   object 
 1   NetWorth in billions  2755 non-null   float64
 2   Country               2755 non-null   object 
 3   Source                2755 non-null   object 
 4   Rank                  2755 non-null   int64  
 5   Age                   2755 non-null   int64  
 6   Industry              2755 non-null   object 
dtypes: float64(1), int64(2), object(4)
memory usage: 150.8+ KB


#### <span style="color:crimson">Q3. What is the data type of the variable Source? Object</span>

### Checking for missing values

In [6]:
df.isnull().sum()

Name                    0
NetWorth in billions    0
Country                 0
Source                  0
Rank                    0
Age                     0
Industry                0
dtype: int64

#### <span style="color:crimson">Q4. How many missing values exists in the industry feature? 0</span>

### Print the basic stats for this dataset

In [7]:
df.describe()

NetWorth in billions         Rank          Age
count           2755.000000  2755.000000  2755.000000
mean               4.749220  1345.663521    63.014519
std                9.615358   772.669811    13.259017
min                1.000000     1.000000    18.000000
25%                1.500000   680.000000    54.000000
50%                2.300000  1362.000000    62.000000
75%                4.200000  2035.000000    72.000000
max              177.000000  2674.000000    97.000000

#### <span style="color:crimson">Q5. Who is the yongest billionare in this list? 18, Kevin David Lehmann</span>

18, Kevin David Lehmann    

### Sort the dataset  by Age (ascending) and reset the index

In [8]:
df.sort_values('Age', ascending=True, inplace=True)
df.reset_index(drop=True, inplace=True)
df.head(5)

Name  NetWorth in billions        Country       Source  \
0  Kevin David Lehmann                   3.3        Germany   drugstores   
1   Alexandra Andresen                   1.4         Norway  investments   
2          Wang Zelong                   1.5          China    chemicals   
3   Katharina Andresen                   1.4         Norway  investments   
4       Austin Russell                   2.4  United States      Unknown   

   Rank  Age          Industry  
0   925   18  Fashion & Retail  
1  2141   24       Diversified  
2  2035   24   Metals & Mining  
3  2141   25       Diversified  
4  1299   26        Automotive

In [9]:
# run
df.iloc[1200,3]

'sensor technology'

#### <span style="color:crimson">Q6. What is the output of `df.iloc[1200,3]`? sensor technology</span>

sensor technology    

### Create a new column `Year_of_birth`
Note: The dataset is from the year 2020. The age of those Billionaires is as of 2020.

In [10]:
# create a new column 'Year_of_birth'
df['Year_of_birth'] = 2020 - df['Age']


In [11]:
#view the data
df


Name  NetWorth in billions        Country  \
0     Kevin David Lehmann                   3.3        Germany   
1      Alexandra Andresen                   1.4         Norway   
2             Wang Zelong                   1.5          China   
3      Katharina Andresen                   1.4         Norway   
4          Austin Russell                   2.4  United States   
...                   ...                   ...            ...   
2750        Masatoshi Ito                   4.0          Japan   
2751    S. Daniel Abraham                   2.0  United States   
2752       Charles Munger                   2.0  United States   
2753        David Murdock                   2.1  United States   
2754          Robert Kuok                  12.6       Malaysia   

                            Source  Rank  Age               Industry  \
0                       drugstores   925   18       Fashion & Retail   
1                      investments  2141   24            Diversified   
2                        chemicals  2035   24        Metals & Mining   
3                      investments  2141   25            Diversified   
4                          Unknown  1299   26             Automotive   
...                            ...   ...  ...                    ...   
2750                        retail   727   96       Fashion & Retail   
2751                     Slim-Fast  1580   96        Food & Beverage   
2752            Berkshire Hathaway  1580   97  Finance & Investments   
2753             Dole, real estate  1517   97        Food & Beverage   
2754  palm oil, shipping, property   171   97            Diversified   

      Year_of_birth  
0              2002  
1              1996  
2              1996  
3              1995  
4              1994  
...             ...  
2750           1924  
2751           1924  
2752           1923  
2753           1923  
2754           1923  

[2755 rows x 8 columns]

### Select Billionaires born in 1996

In [12]:
# Using where condition
df.where(df['Year_of_birth'] == 1996).dropna()

Name  NetWorth in billions Country       Source    Rank  \
1  Alexandra Andresen                   1.4  Norway  investments  2141.0   
2         Wang Zelong                   1.5   China    chemicals  2035.0   

    Age         Industry  Year_of_birth  
1  24.0      Diversified         1996.0  
2  24.0  Metals & Mining         1996.0

#### <span style="color:crimson">Q7. How many Billionaires were born in 1996? 2</span>